src/controllers/DomainController.ts

```diff
@@ -10,13 +10,27 @@ import { DomainMetadataConfig } from "../utils/domainMetadata"
 const env = process.env.ENVIRONMENT || ""
 const environment = (env.toLowerCase() as Environment) || Environment.MAINNET
 export const DomainsController = {
-  domains: function (request: FastifyRequest<{}>, reply: FastifyReply): void {
+  domainsMetadata: function (request: FastifyRequest<{}>, reply: FastifyReply): void {
     const metadata = DomainMetadataConfig[environment]
     if (metadata) {
-      void reply.status(200).send(metadata)
+      void reply.status(200).send(JSON.stringify(metadata))
     } else {
       logger.error(`Unable to find metadata for environment ${environment}`)
       void reply.status(500)
     }
   },
+  resources: function (request: FastifyRequest<{ Params: { domainID: string } }>, reply: FastifyReply): void {
+    const {
+      params: { domainID },
+    } = request
+    const domains = DomainMetadataConfig[environment]
+    if (!domains) {
+      logger.error(`Unable to find metadata for environment ${environment}`)
+      void reply.status(500)
```

Comment:
I would argue this is not 500, as it can only happen if you provide wrong env, right?

Reply:
will change it to 404, since basically you are not finding anything

Reply:
also even if you don't define an environment, it defaults to mainnet one